Mathlib/Data/Set/Function.lean

```diff
@@ -769,6 +780,22 @@ theorem _root_.Disjoint.image {s t u : Set α} {f : α → β} (h : Disjoint s t
   rw [← hf.image_inter hs ht, h, image_empty]
 #align disjoint.image Disjoint.image
 
+lemma InjOn.image_diff {t : Set α} (h : s.InjOn f) : f '' (s \ t) = f '' s \ f '' (s ∩ t) := by
+  refine subset_antisymm (subset_diff.2 ⟨image_subset f (diff_subset s t), ?_⟩)
+    (diff_subset_iff.2 (by rw [← image_union, inter_union_diff]))
+  exact Disjoint.image disjoint_sdiff_inter h (diff_subset _ _) (inter_subset_left _ _)
+
+lemma InjOn.image_diff_of_subset {f : α → β} {t : Set α} (h : InjOn f s) (hst : t ⊆ s) :
+    f '' (s \ t) = f '' s \ f '' t := by
+  rw [h.image_diff, inter_eq_self_of_subset_right hst]
+
+theorem InjOn.imageFactorization_injective (h : InjOn f s) :
+    Injective (s.imageFactorization f) :=
```

Comment:
Is this an iff?

Reply:
Maybe, but applying this in the other direction feels contrived to me.

Reply:
Please avoid creating consensus by closing review comments you didn't address. Maintainers can always decide whether those comments are worth discarding.

Reply:
> Maybe, but applying this in the other direction feels contrived to me. 

Iffs do not necessarily need to be applied the other way to be useful. A great use of them is rewriting.

Reply:
You lose dot notation this way. Adding a separate `iff` lemma is possible, but it feels almost useless to me.

Reply:
Two lemmas sounds like the way to go to me